channels/rdpgfx/client/rdpgfx_main.c

```diff
@@ -218,7 +217,7 @@ static UINT rdpgfx_recv_caps_confirm_pdu(RDPGFX_CHANNEL_CALLBACK* callback,
 	}
 
 	Stream_Read_UINT32(s, capsSet.version); /* version (4 bytes) */
-	Stream_Read_UINT32(s, capsDataLength); /* capsDataLength (4 bytes) */
+	Stream_Read_UINT32(s, capsSet.length); /* capsDataLength (4 bytes) */
```

Comment:
for current use most likely ok, but with capssets that have differing lengths `gfx->ConnectionCaps` should be adjusted to be a dynamically allocated copy.